### ISSUE TITLE
ZAP does not find an obvious SQL injection flaw

### ISSUE THREAD

**Issue by alexandersolovey** [zaproxy/zaproxy#1851]
Hi all,

I tried ZAP and it seems to have troubles to identify a pretty obvious SQL injection flaw.

I have an application that I created intentionally vulnerable. It uses Spring authentication. I tried ZAP on it and it finds the vulnerable page but it does not find the vulnerability. What do I need to change to configure it so that it finds the SQL injection flaw?

Thanks,
Alexander

**Comment by kingthorin** [comment]
Did you have the SQL Injection scanner enabled in the scan policy? 
Did your policy have a DB related technology selected/specified?

**Comment by alexandersolovey** [comment]
Yes, I set up the threshold to 'high' and strength to 'insane' for the SQL Injection test in the "Scan Policy..." under the "Analyze" menu. It performs a lot of tests for the j_spring_security_check page including some SQL Injection but when it reaches to the page I am interested in (it is called /unsafe) it just gets the page and does not do any tests on it. I am obviously new to ZAP but I am following a tutorial and I think I am doing everything I should.

Thanks,
Alexander

**Comment by psiinon** [comment]
Have you tried the beta active scan rules?
https://github.com/zaproxy/zap-extensions/wiki/HelpAddonsAscanrulesBetaAscanbeta
And the 'sqlplugin' ? (no help for this one I'm afraid)
they can both be downloaded from the ZAP Marketplace.

**Comment by kingthorin** [comment]
@alexandersolovey are you able to provide the example and setup info?

**Comment by pedropamn** [comment]
Same here... I'm scanning a Damn app in low security level (http://www.dvwa.co.uk/)

**Comment by kingthorin** [comment]
@pedropamn can you read through the previous entries in the ticket and provide further details?

**Comment by kingthorin** [comment]
@alexandersolovey are you able to provide the example and setup info for the original issue?
